lib/schema.graphql

```diff
@@ -0,0 +1,98 @@
+directive @aws_cognito_user_pools(
+    cognito_groups: [String!]
+) on FIELD_DEFINITION | OBJECT
+
+type Product {
+    Id: ID!
+    Name: String!
+    Description: String
+    Price: Float!
+    Currency: String!
+    Weight: Int!
+    ImageUrl: String
+    Supplier: String
+    Category: ID!
+}
+
+type ProductCategory {
+    Id: ID!
+    Name: String!
+    Description: String
+    Products: [Product]
+}
+
+type Query {
+    listCategories: [ProductCategory]
+    getCategoryById(Id: ID!): ProductCategory
+    listProductsByCategory(Category: ID!): [Product]
+    getProductById(Id: ID!): Product
+}
+
+input ProductInput {
+    Name: String!
+    Description: String
+    Price: Float!
+    Currency: String!
+    Weight: Int!
+    ImageUrl: String
+    Supplier: String
+    Category: ID!
+}
+
+input CategoryInput {
+    Name: String!
+    Description: String
+}
+
+type Order {
+    Id: ID!
+    CustomerEmail: String!
+    ShippingAddress: String!
+    OrderDetails: [OrderDetail]!
+    OrderDate: String!
+}
+
+type OrderDetail {
+    ProductId: ID!
+    Quantity: Int!
+}
+
+input OrderInput {
+    ShippingAddress: String!
+    OrderDetails: [OrderDetailInput]!
+}
+
+input OrderDetailInput {
+    ProductId: ID!
+    Quantity: Int!
+}
+
+type Mutation {
+    addProduct(product: ProductInput!): Product @aws_cognito_user_pools(cognito_groups: ["Admin"])
+    addCategory(category: CategoryInput!): ProductCategory @aws_cognito_user_pools(cognito_groups: ["Admin"])
+    updateProduct(
+        Id: ID!
+        Name: String
+        Description: String
+        Price: Float
+        Currency: String
+        Weight: Int
+        ImageUrl: String
+        Supplier: String
+        Category: ID
+    ): Product @aws_cognito_user_pools(cognito_groups: ["Admin"])
+    updateCategory(
+        Id: ID!
+        Name: String
+        Description: String
+    ): ProductCategory @aws_cognito_user_pools(cognito_groups: ["Admin"])
+    deleteProduct(Id: ID!): Boolean @aws_cognito_user_pools(cognito_groups: ["Admin"])
+    deleteCategory(Id: ID!): Boolean @aws_cognito_user_pools(cognito_groups: ["Admin"])
+    placeOrder(order: OrderInput!): String @aws_cognito_user_pools(cognito_groups: ["User"])
```

Comment:
I'm still able to create/delete/update Products/Categories through 'Customer' group account.
`placeOrder` doesn't exist in appsync. (no 'User' cognito_group).

Reply:
Challenge was with using @aws_cognito_user_pools. Used @aws_auth instead.

When using Cognito as your only authorization provider with AppSync, opt for @aws_auth for reliable group-based authorization, not @aws_cognito_user_pools. In contrast, if Cognito is one of multiple authorization providers, @aws_cognito_user_pools should be used, as it effectively manages group-based authorization in these scenarios.